src/odemis/acq/acqmng.py

```diff
@@ -726,11 +726,31 @@ def __init__(self, microscope: model.HwComponent, components: Set[model.HwCompon
                     continue
                 # Store current value of VA (calling .value might take some time)
                 self._all_settings[comp.name][va_name] = [va.value, va.unit]
+
+                if va_name == "position" and hasattr(comp, "axes") and isinstance(comp.axes, dict):
+                    # For the .position, the axes definition may contain also the "user-friendly" name
+                    # of the axis, so we can add it to the metadata.
+                    def update_settings(value: Dict[str, float], comp_name=comp.name, va_name=va_name, comp=comp):
+                        try:
+                            axes_def: Dict[str, model.Axis] = comp.axes
+                            value = value.copy()
+                            for axis_name, p in value.items():
+                                if (axis_name in axes_def and hasattr(axes_def[axis_name], "choices")
+                                    and isinstance(axes_def[axis_name].choices, dict)
+                                    and p in axes_def[axis_name].choices
+                                   ):
+                                    value[axis_name] = f"{p} ({axes_def[axis_name].choices[p]})"
+                        except Exception:
+                            logging.exception("Failed to interpret user-friendly axis info for %s.position", comp_name)
+
+                        self._all_settings[comp_name][va_name][0] = value
```

Comment:
I do not see anything under if contidition, is line 746 indexed properly?

Reply:
Yes. Depending on the if/else the `update_settings()` function is different.